src/features/class-parsely-support.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Parsely_Support class file
+ *
+ * @package wp-curate
+ */
+
+namespace Alley\WP\WP_Curate\Features;
+
+use Alley\WP\Types\Feature;
+use Parsely\RemoteAPI\Analytics_Posts_API;
+
+/**
+ * Add support for Parsely, if the plugin is installed.
+ */
+final class Parsely_Support implements Feature {
+	/**
+	 * Set up.
+	 */
+	public function __construct() {}
+
+	/**
+	 * Boot the feature.
+	 */
+	public function boot(): void {
+		if ( ! class_exists( 'Parsely\Parsely' ) ) {
+			return;
+		}
+		// Elsewhere in the plugin, we'll use $GLOBALS['parsely'], but it is not available here.
+		$parsely = new \Parsely\Parsely();
+		// If we don't have the API secret, we can't use the Parsely API.
+		if ( ! $parsely->api_secret_is_set() ) {
+			return;
+		}
+		add_filter( 'wp_curate_use_parsely', '__return_true' );
+		add_filter( 'wp_curate_trending_posts_query', [ $this, 'add_parsely_trending_posts_query' ], 10, 2 );
+	}
+
+	/**
+	 * Gets the trending posts from Parsely.
+	 *
+	 * @param array<number>        $posts The posts, which should be an empty array.
+	 * @param array<string, mixed> $args The WP_Query args.
+	 * @return array<number> Array of post IDs.
+	 */
+	public function add_parsely_trending_posts_query( array $posts, array $args ): array {
+		$parsely = $GLOBALS['parsely'];
+		if ( ! $parsely->api_secret_is_set() ) {
+			return $posts;
+		}
+		$trending_posts = $this->get_trending_posts( $args );
+		return $trending_posts;
+	}
+
+	/**
+	 * Gets the trending posts from Parsely.
+	 *
+	 * @param array<string, mixed> $args The WP_Query args.
+	 * @return array<number> An array of post IDs.
+	 */
+	public function get_trending_posts( array $args ): array {
+		// TODO: Add failover if we're not on production.
+		/**
+		 * Filter the period start for the Parsely API.
+		 *
+		 * @param string $period_start The period start.
+		 * @return string The period start.
+		 */
+		$period_start = apply_filters( 'wp_curate_parsely_period_start', '1d' );
+		$parsely_args = [
+			'limit'        => $args['posts_per_page'],
+			'sort'         => 'views',
+			'period_start' => $period_start,
+			'period_end'   => 'now',
+		];
+		if ( isset( $args['tax_query'] ) && is_array( $args['tax_query'] ) ) {
+			foreach ( $args['tax_query'] as $tax_query ) {
+				if ( isset( $tax_query['taxonomy'] ) && 'category' === $tax_query['taxonomy'] ) {
+					$parsely_args['section'] = implode( ', ', $this->get_slugs_from_term_ids( $tax_query['terms'], $tax_query['taxonomy'] ) );
```

Comment:
Is it hardcoded in the Parsely plugin that `section` corresponds to `category`? Could it correspond to other taxonomies? 

Also, does this implementation assume that the official Parsely plugin is being used and not a custom integration? That seems fair to assume, but it's worth documenting.

Reply:
I have added some logic to handle Parsely options better.

we check for the `Parsely\Parsely` class, so it won't work without the plugin installed. I'll need to write some docs on this feature at some point.